include/pybind11/detail/internals.h

```diff
@@ -83,9 +83,9 @@ template <typename value_type>
 using type_map = std::unordered_map<std::type_index, value_type, type_hash, type_equal_to>;
 
 struct override_hash {
-    inline size_t operator()(const std::pair<const PyObject *, const char *>& v) const {
+    inline size_t operator()(const std::pair<const PyObject *, std::string>& v) const {
         size_t value = std::hash<const void *>()(v.first);
-        value ^= std::hash<const void *>()(v.second) + 0x9e3779b9 + (value<<6) + (value>>2);
+        value ^= std::hash<std::string>()(v.second);
```

Comment:
From Boris:

> Also, also, the changes to the hash function are really wrong. You don't want to combine hashes with just an xor.

Reply:
Hm... Should I keep the original pattern then?
```
value ^= name_hash + 0x9e3779b9 + (value<<6) + (value>>2)
```
(is there a good reference you'd suggest for this?)

Reply:
Ah: https://www.variadic.xyz/2018/01/15/hashing-stdpair-and-stdtuple/

Reply:
Done in 7e991aa